sklearn/decomposition/dict_learning.py

```diff
@@ -112,12 +112,20 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         try:
             err_mgt = np.seterr(all='ignore')
 
+            # TODO: Should verbose argument be passed to this?
+            if positive:
+                raise ValueError(
+                    """
+                    Positive constraint not supported for \"lasso_lars\"
```

Comment:
Actually, `positive` was not deprecated in `lasso_lars` for what I can see in the deprecation message.
@agramfort could you confirm which class is affected by the `positive` keyword deprecation?

Reply:
`LassoLars` class inherits from `Lars` class (only), so when I removed `positive` from the `Lars` class, I removed it from `LassoLars` as well.

I looked at the source code, searching for LassoLars reimplementation of the `positive` parameter and couldn't find any.

However, as I am a pretty recent contributor, I may have missed something and my comment may need a double check.

Reply:
I spoke with @agramfort IRL and we should not remove the `positive` parameter from the `LassoLars` since we force it to use `lasso` as a method (this is hardcoded in the class).

I already implemented the trick in the previous PR which I closed #13782 
The idea is that you can set a variable `_positive` to be `False` for the `Lars` class and not expose the parameter:

https://github.com/scikit-learn/scikit-learn/pull/13782/files#diff-bdf3cc726bb0cd772cd2ffd870e2ece8R851

Then you can check the `_fit` method as well.

Reply:
I re-added the positive parameter in lasso-lars algorithm.

I actually set the variable `positive` to `False` in `Lars` class (and not `_positive`).

Using a `_positive` variable would have forced me to use `self._positive` in every algorithm implemented in `Lars`. As `LassoLars` inherits from `Lars`, I would have had to do someting like

```
class LassoLars(Lars):
     def __init____(self, [ ... ], positive):
        ...
        self._positive = positive
```